### PR TITLE
[Tests-Only] Update phpstan to latest version

### PR DESCRIPTION
**Pull request by dpakach**
Related - owncloud/QA#642

Update phpstan to 0.12

### CODE REVIEW

File: lib/Controller/DocumentController.php [hunk 1]
```diff
@@ -1084,7 +1085,7 @@ private function putRelative($fileId, $owner, $editor, $suggested) {
 		$serverHost = $this->request->getServerProtocol() . '://' . $this->request->getServerHost();
 
 		// Continue editing
-		$attributes = WOPI::ATTR_CAN_VIEW | WOPI::ATTR_CAN_UPDATE | WOPI::ATTR_CAN_DOWNLOAD | WOPI::ATTR_CAN_PRINT;
+		$attributes = WOPI::ATTR_CAN_VIEW | WOPI::ATTR_CAN_UPDATE | WOPI::ATTR_CAN_PRINT;
```
Comment (dpakach) [inline lib/Controller/DocumentController.php:1088]: `WOPI::ATTR_CAN_DOWNLOAD` is undefined and not so different than `WOPI::ATTR_CAN_VIEW` IMO

File: lib/Http/DownloadResponse.php [hunk 2]
```diff
@@ -50,11 +53,11 @@ public function render() {
 					$this->sendNotSatisfiable($size);
```
Comment (dpakach) [inline lib/Http/DownloadResponse.php:53]:
```
					$this->sendNotSatisfiable($size);
```
This method is not defined and causing the phpstan to fail. not sure how to fix this

File: lib/Http/DownloadResponse.php [hunk 3]
```diff
@@ -50,11 +53,11 @@ public function render() {
...
 				}
 				if ($parts[0]==='') {
-					$start = $size - $parts[1];
+					$start = $size - strlen($parts[1]);
```
Comment (dpakach) [inline lib/Http/DownloadResponse.php:56]: `-` operation between string and int fails in phpstan.

**Comment by dpakach** [comment]
There are still calls to some undefined methods causing phpstan to fail in this repo, not sure how to fix those.

@phil-davis  please review

File: lib/Http/DownloadResponse.php [hunk 4]
```diff
@@ -47,18 +50,18 @@ public function render() {
 				$parts = \explode('-', $range);
 
 				if ($parts[0]==='' && $parts[1]=='') {
-					$this->sendNotSatisfiable($size);
+					$this->sendRangeNotSatisfiable($size);
```
Comment (phil-davis) [inline lib/Http/DownloadResponse.php:53]: This fixes a crash that would happen if a wrong range is requested.

File: lib/Http/DownloadResponse.php [hunk 5]
```diff
@@ -47,18 +50,18 @@ public function render() {
...
 					$end = $size - 1;
 				} else {
 					$start = $parts[0];
-					$end = ($parts[1]==='') ? $size - 1 : $parts[1];
+					$end = ($parts[1]==='') ? $size - 1 : (int)$parts[1];
 				}
 
 				if ($start > $end) {
-					$this->sendNotSatisfiable($size);
+					$this->sendRangeNotSatisfiable($size);
```
Comment (phil-davis) [inline lib/Http/DownloadResponse.php:64]: This fixes a crash that would happen if a wrong range is requested.

File: lib/Controller/DocumentController.php (hunk 1 above)
Comment (phil-davis) [reply lib/Controller/DocumentController.php:1088]: I raised issue #335

**Pull request merged**